### PR TITLE
Refactor and comment the locale files.

### PR DESCRIPTION
**Pull request by viroulep**
Added context comment to the locale files: these comments will be displayed by @jonatanklosko  "Internationalize" app, and will appear on a "key by key" basis, therefore the singular is used in these comments.

Commenting them showed that some keys could be reordered or grouped, so this has been done as well.

This PR will conflict with @pingskills #1001 (on the `en.yml` locale file), so maybe it's a good idea to wait until his PR is merged.

### CODE REVIEW

**Comment by jonatanklosko** [comment]
I've just had a glance at how this looks in the app and it's really awesome and useful! Great work with adding all those comments Philippe =)

**Comment by jonatanklosko** [comment]
Two screenshoots to show you all the idea:

screenshot from 2016-12-31 13-39-44

screenshot from 2016-12-31 13-43-00

**Comment by larspetrus** [comment]
Where can I learn more about this i...ze app?

File: WcaOnRails/config/locales/en.yml [hunk 1]
```diff
@@ -70,12 +74,15 @@ en:
...
       invalid: "invalid"
+    #context: Key used for the pluralization of "x day(s)" in a sentence
     days:
       one: "1 day"
       other: "%{count} days"
+    #context: Key used for the pluralization of "x event(s)" in a sentence
     these_events:
       one: "this event"
       other: "these %{count} events"
+  #context: Key used in the Oauth part of the site (checkout the /oauth/applications page for more details)
```
Comment (jfly) [inline WcaOnRails/config/locales/en.yml]: nit: Oauth -> OAuth

**Comment by jonatanklosko** [comment]
@larspetrus the app lives [here](http://internationalize.herokuapp.com/), where you can find a short, general description. Also the code is on my GitHub.

File: WcaOnRails/config/locales/en.yml (hunk 1 above)
Comment (viroulep) [inline WcaOnRails/config/locales/en.yml]: fixed :)

**Comment by viroulep** [comment]
@jfly:
> Are we supposed to fully specify i18n keys now, or that a tangentially related change?

Actually no: when commenting the locale I realized it made sense to put the exact same context comment on several sibling nodes, such as on `competitions.index`, `competitions.index_competitions_list`, etc.
So I grouped them under the "main" view they were related to (here `competitions.index`).

I think it's fine to use relative keys when possible, but I'd rather group the keys that appear on the same view all together (which means keys in partials would be defined in their "parent" view, and there it would be necessary to fully specify the keys).

I hope my reasoning makes sense!

**Comment by jfly** [comment]
LGTM! I'd support merging this up sooner than later (either way, someone has to deal with conflicts, and having this PR wait when it's ready to merge seems like a lose), but please talk to @pingskills and figure out a schedule for you guys to merge up your respective work.

**Comment by pingskills** [comment]
@viroulep if this is ready then go ahead and merge it up. Then if I need any help merging my changes I'll reach out to you.

**Comment by viroulep** [comment]
@pingskills ok then I'll merge this!

FYI I just tried merging your work in this and the conflict is pretty simple: it's a matter of keeping a context comment on its correct node, which shouldn't be hard ;)

**Pull request merged**